server/chat-plugins/rock-paper-scissors.ts

```diff
@@ -0,0 +1,426 @@
+		return this.playerTable[user.id];
+	}
+}
+
+function getGame(context: CommandContext) {
+	const room = context.requireRoom();
+	const game = room.getGame(RPSGame);
+	if (!game) throw new Chat.ErrorMessage(`There is no Rock, Paper, Scissors game going on in this room.`);
+	if (game.gameid !== 'rockpaperscissors') {
+		throw new Chat.ErrorMessage(`A game of ${game.title} is already going on.`);
+	}
+	return game;
+}
+
+function findExisting(user1: string, user2: string) {
+	return Rooms.get(`rps-${user1}-${user2}`) || Rooms.get(`rps-${user2}-${user1}`);
+}
+
+export const commands: ChatCommands = {
+	rps: 'rockpaperscissors',
+	rockpaperscissors: {
+		challenge: 'create',
+		create(target, room, user) {
+			target = target.trim();
+			const [rounds, userid] = target.split(',');
+			if (!rounds || !target) {
+				return this.parse('/help rps');
+			}
+			const roundsNum = parseInt(target);
+			if (isNaN(roundsNum)) {
+				return this.errorReply(`Invalid round number.`);
+			}
+			if (roundsNum > MAX_ROUNDS) {
+				return this.errorReply(`You cannot set more than ${MAX_ROUNDS} rounds.`);
+			}
+			const targetUser = userid ? Users.get(userid) : this.pmTarget;
+			if (targetUser === user) return this.errorReply(`You cannot challenge yourself.`);
+			if (!targetUser) {
+				return this.errorReply(
+					`User ${this.targetUsername} not found. Either specify a username or use this command in PMs.`
+				);
+			}
+			const existingRoom = findExisting(user.id, targetUser.id);
+			const roomid = existingRoom ? existingRoom.roomid : `rps-${user.id}-${targetUser.id}`;
+			if (existingRoom?.game && !existingRoom.game.ended) {
+				return this.errorReply(`You already have a Rock, Paper, Scissors game against ${targetUser.name}.`);
+			}
+			if (!this.pmTarget) this.pmTarget = targetUser;
+			challengeRequests.set(targetUser.id, [user.id, rounds]);
+			this.sendChatMessage(
+				`/raw ${user.name} challenged you to Rock, Paper, Scissors!` +
+				`<button class="button" name="send" value="/j ${roomid}"><strong>Accept</strong></button></div>`,
+			);
+		},
+
+		accept(target, room, user) {
+			const request = challengeRequests.get(user.id);
+			if (!request) return this.errorReply(`You have no Rock, Paper, Scissors request pending from ${target}.`);
+			const [id, rounds] = request;
+			const targetUser = Users.get(id)!;
+			const existingRoom = findExisting(user.id, targetUser.id);
+			const options = {
+				modchat: '+',
+				isPrivate: true,
+			};
+			const roomid = `rps-${targetUser.id}-${user.id}`;
+			const gameRoom = existingRoom ? existingRoom : Rooms.createGameRoom(
+				roomid as RoomID, `[RPS] ${user.name} vs ${targetUser.name}`, options
+			);
+			gameRoom.game = new RPSGame(gameRoom, parseInt(rounds));
+			gameRoom.add(
+				`|raw|<h2>Welcome to the Rock, Paper, Scissors game of ${user.name} vs ${targetUser.name}!</h2>` +
+				`Use /rps start to start the game, once both players have joined!`
+			).update();
```

Comment:
Do you need a start command? Why can't it just start instantly?

If there's a need for it to start non-instantly, it should probably at least be a button rather than telling you to use a command.

Reply:
(Design standard 3)

Reply:
You marked this as resolved but this message is still here. It looks like you're now autostarting the game. Is this still necessary? Do we need an `/rps start` command at all?

Reply:
Oh, yeah, i'll remove that. forgot that.